src/governance.cpp

```diff
@@ -921,37 +930,17 @@ bool CGovernanceObject::IsValidLocally(const CBlockIndex* pindex, std::string& s
 
 CAmount CGovernanceObject::GetMinCollateralFee()
 {
-    CAmount nMinFee = 0;
     // Only 1 type has a fee for the moment but switch statement allows for future object types
     switch(nObjectType) {
-    case GOVERNANCE_OBJECT_PROPOSAL:
-        nMinFee = GOVERNANCE_PROPOSAL_FEE_TX;
-        break;
-    case GOVERNANCE_OBJECT_TRIGGER:
-        nMinFee = 0;
-        break;
-    default:
-      {
-        std::ostringstream ostr;
-        ostr << "CGovernanceObject::GetMinCollateralFee ERROR: Unknown governance object type: " << nObjectType;
-        throw std::runtime_error(ostr.str());
-      }
+        case GOVERNANCE_OBJECT_PROPOSAL:    return GOVERNANCE_PROPOSAL_FEE_TX;
+        default:                            return 0;
```

Comment:
"There is more logic in than just that in CGovernanceObject::IsValidLocally already."

I don't understand that statement.  Yes IsValidLocally contains a lot of logic but it doesn't really have a reason to check that the object type is known (unless it's a trigger).

I actually think this code should be refactored to use inheritance properly rather than having logic dependent on type codes but I don't think it makes sense to do that now.  It's better to wait until we have the requirements for 12.2/dashdrive/evo.